plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/GlueIcebergUtil.java

```diff
@@ -38,18 +48,86 @@ public static TableInput getTableInput(String tableName, Optional<String> owner,
     {
         ImmutableMap.Builder<String, String> tableParameters = ImmutableMap.<String, String>builder()
                 .putAll(parameters)
-                .put(METADATA_LOCATION_PROP, metadataLocation);
+                .put(METADATA_LOCATION_PROP, metadataLocation)
+                .put(COLUMNS_INFO_VALID_FOR_METADATA_LOCATION, metadataLocation);
```

Comment:
This additional property is so that we detect when metadata is updated without updating the table columns.
(ideally Spark Iceberg would do same approach as we do, but doesn't have to)